src/media/backends/azure-backend.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Inject, Injectable } from '@nestjs/common';
+import mediaConfiguration, { MediaConfig } from '../../config/media.config';
+import { ConsoleLoggerService } from '../../logger/console-logger.service';
+import { MediaBackend } from '../media-backend.interface';
+import { BackendData } from '../media-upload.entity';
+import {
+  BlobServiceClient,
+  BlockBlobClient,
+  ContainerClient,
+} from '@azure/storage-blob';
+import { BackendType } from './backend-type.enum';
+
+@Injectable()
+export class AzureBackend implements MediaBackend {
+  private config: MediaConfig['backend']['azure'];
+  private client: ContainerClient;
+
+  constructor(
+    private readonly logger: ConsoleLoggerService,
+    @Inject(mediaConfiguration.KEY)
+    private mediaConfig: MediaConfig,
+  ) {
+    this.logger.setContext(AzureBackend.name);
+    this.config = mediaConfig.backend.azure;
+    if (mediaConfig.backend.use === BackendType.AZURE) {
+      // only create the client if the backend is configured to azure
+      const blobServiceClient = BlobServiceClient.fromConnectionString(
+        this.config.connectionString,
+      );
+      this.client = blobServiceClient.getContainerClient(this.config.container);
```

Comment:
You could call createIfNotExists on the container client to auto-create the container (with access set to "blob").

Reply:
Is that common in the Azure world? I'm not to sure I want to create possible typo containers instead of failing, but if it's rather strange to not create the container that's how we should do it, too.

Reply:
That's something we always did to reduce manual steps when deploying, but I don't know if that is common or not. It helps to create the correct kind of container policy.

Reply:
We talked about this and general opinion was, we shouldn't do this…

Reply:
Sure, just an idea.